### DISCUSSION TITLE
XEM Emulator

### DISCUSSION THREAD

**Discussion by LeeThompson** [pymedusa/Medusa#9947] [Ideas & Feature requests]
Since being 'forced' to switch to tvmaze, I've had to manually make a lot of scene exceptions (that and the overall data quality on tvmaze is not very good).

This appears to be, in part, that tvmaze isn't supported by XEM; would it be possible for medusa to apply XEM lookups for non thetvdb/anidb indexed shows by using some level of emulation? (i.e. maybe somehow get the thetvdb show id and use that etc).

It would make having to use tvmaze suck less.

(I realize this is partially similar to #9725 but this would be a larger scale workaround.)

Alternately, do our (Medusa) own scene exception list.    (It could be done agnostic to indexing service since really what you're trying to do is eliminate false positives, so the system could allow different pieces of information such as airdates, season episode counts, episode titles all used to attempt ensure the correct show is being matched - in lieu of a showid from one of the services.)

**Comment by p0psicles** [comment]
IT's very easy to implement. But there is one issue. What if the indexers don't follow the same season/episode structure. You would asume that a season scene_exception for tvdb is the same as for tvmaze. Maybe we could add it as an option.

@medariox what do you think?

**Reply by p0psicles** [reply]
You could play around with this branch: `feature/map-indexer-for-exceptions`
PR #9949

**Reply by medariox** [reply]
I agree that it's a bit risky to assume the same structure... Maybe we can add a check and only use it for shows with the same structure?

**Reply by p0psicles** [reply]
I though about that. But the thing witt that is, is that it's not 100% clear when a structure is the same.
For example what to do if tvdb add another episode for that week, and tvdb hasn't yet. So that can cause all kind of issues that users can come up with.
